src/main/java/org/kaazing/netx/http/internal/HttpURLConnectionHandler.java

```diff
@@ -250,76 +248,85 @@ public InputStream getInputStream() throws IOException {
 
             switch (state) {
             case HANDSHAKE_SENT:
-                input = socket.getInputStream();
+                LineReader reader = null;
 
-                BufferedReader reader = new BufferedReader(new InputStreamReader(input, "US-ASCII"));
-                String start = reader.readLine();
-                connection.addHeaderField(null, start);
+                try {
+                    input = socket.getInputStream();
 
-                if (start == null) {
-                    throw new IllegalStateException("Bad HTTP/1.1 syntax");
-                }
-                Matcher startMatcher = PATTERN_START.matcher(start);
-                if (!startMatcher.matches()) {
-                    throw new IllegalStateException("Bad HTTP/1.1 syntax");
-                }
-                int responseCode = parseInt(startMatcher.group(1));
-                String responseMessage = startMatcher.group(2);
-                connection.setResponse(responseCode, responseMessage);
-
-                Map<String, List<String>> cookies = null;
-                List<String> challenges = null;
-                for (String header = reader.readLine(); !header.isEmpty(); header = reader.readLine()) {
-                    int colonAt = header.indexOf(':');
-                    if (colonAt == -1) {
+                    reader = new LineReader(input);
+
+                    String start = reader.readLine();
+                    connection.addHeaderField(null, start);
+
+                    if ((start == null) || start.isEmpty()) {
```

Comment:
`BufferedReader.readLine()` returns null for EOF, not empty string.  Let's make `LineReader` consistent with `BufferedReader`.